client.go

```diff
@@ -44,13 +44,19 @@ func SelectOneOf(protos []string, rwc io.ReadWriteCloser) (string, error) {
 }
 
 func handshake(rwc io.ReadWriteCloser) error {
-	if err := delimWrite(rwc, []byte(ProtocolID)); err != nil {
-		return err
-	}
+	errCh := make(chan error)
```

Comment:
Thoughts on making this buffered? I know theres no early outs, but if `ReadNextToken` panics for some reason it could leave a hanging goroutine. 
Probably not a big deal though

Reply:
We don't catch the panic, so we'll just crash the program anyways.

Reply:
Oh, *I* see. Hm. I'm always wary about recovering from panics, but you're probably right.

Reply:
Fixed: https://github.com/multiformats/go-multistream/pull/30